akvo/rsr/management/commands/switch_indicators_to_cumulative_reporting.py

```diff
@@ -0,0 +1,59 @@
+import argparse
+from io import TextIOBase
+from typing import List
+from django.core.management.base import BaseCommand
+from django.db import transaction
+from akvo.rsr.models import Project, Indicator
+from akvo.rsr.models.result.utils import QUANTITATIVE
+
+
+class Command(BaseCommand):
+    help = "Switch all quantitative indicators of the given project and its descendants to cumulative reporting"
+
+    def add_arguments(self, parser: argparse.ArgumentParser):
+        parser.add_argument("project_id", type=int)
+        parser.add_argument(
+            "--exclude", nargs="+", help="ID of indicators that should not be changed"
+        )
+        parser.add_argument(
+            "--dry-run", action="store_true", help="Don not actually apply the changes"
+        )
+
+    def handle(self, *args, **options):
+        project = Project.objects.get(id=int(options["project_id"]))
+        excludes = [int(id) for id in options["exclude"]] if options["exclude"] else []
+        runner = CommandRunner(self.stdout, project, excludes, options["dry_run"])
+
+        try:
+            runner.run()
+        except InterruptedError:
+            self.stdout.write("Changes not applied\n")
+        else:
+            self.stdout.write("Changes applied\n")
+        self.stdout.write("DONE!\n")
+
+
+class CommandRunner:
+    def __init__(
+        self, stdout: TextIOBase, project: Project, excludes: List[int], dry_run=False
+    ):
+        self.stdout = stdout
+        self.project = project
+        self.excludes = excludes
+        self.dry_run = dry_run
+
+    @transaction.atomic
+    def run(self):
+        indicators = Indicator.objects.filter(
+            result__project=self.project, type=QUANTITATIVE
+        )
```

Comment:
I just tested this and it doesn't work with descendants that defined their own quantitative indicators. They won't be picked up in the inheritance chain.

A possible fix could be

```python
indicators = Indicator.objects.filter(result__project__in=self.project.descendants, type=QUANTITATIVE)
```

To test that it works, I ran

```sh
./manage.py switch_indicators_to_cumulative_reporting 9062
./manage.py shell
```
```python
from akvo.rsr.models import *
QUANTITATIVE=1
mwc = Projects.objects.get(id=9062)
Indicator.objects.filter(result__project__in=mwc.descendants(), type=QUANTITATIVE, cumulative=False).count())
# 159

# get those without a parent
Indicator.objects.filter(result__project__in=mwc.descendants(), type=QUANTITATIVE, cumulative=False, parent_indicator__isnull=True).count()
# 59
```

They are across multiple projects.

Reply:
Fixed